core/transcoder.go

```diff
@@ -115,23 +115,44 @@ func (nv *NvidiaTranscoder) Transcode(ctx context.Context, md *SegTranscodingMet
 	return resToTranscodeData(ctx, res, out)
 }
 
-// Test which capabilities transcoder supports
-func TestTranscoderCapabilities(devices []string) (caps []Capability, fatalError error) {
+type transcodeTestParams struct {
+	TestAvailable bool
+	Cap           Capability
+	OutProfile    ffmpeg.VideoProfile
+	SegmentPath   string
+}
+
+func (params transcodeTestParams) IsRequired() bool {
+	return InArray(params.Cap, DefaultCapabilities())
+}
+
+func (params transcodeTestParams) Kind() string {
+	if params.IsRequired() {
+		return "required capability"
+	}
+	return "optional capability"
+}
+
+func (params transcodeTestParams) Name() string {
+	name, err := CapabilityToName(params.Cap)
+	if err == nil {
+		return name
+	}
+	return "unknown"
+}
+
+type continueLoop bool
+
+func forEachTranscoderSample(handler func(*transcodeTestParams) continueLoop) {
 	// default capabilities
```

Comment:
Nit: Could it be changed to a more explicit function name?
```suggestion
	func transcodeWithSample(handler func(*transcodeTestParams) continueLoop) 
```

Reply:
fixed